### ISSUE TITLE
payara-micro:bundle ignores finalName

### ISSUE THREAD

**Issue by Skare69** [payara/maven-plugins#13]
## The problem
Bundling a WAR file with the Payara Micro Maven plugin to create an Uber JAR ignores the WAR's `finalName` setting and just uses the GAV parameters. 

This leads to the WAR file being deployed under a different path which in turn leads to problems with any other resources using the `finalName` (e.g. SwaggerUI). 

## Example
**The `pom.xml` file:**
```
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.some-project</groupId>
    <artifactId>some-project</artifactId>
    <version>0.1-SNAPSHOT</version>
    <packaging>war</packaging>
    ...
    <build>
        <finalName>some-project</finalName>
        <plugins>
            ...
            <plugin>
                <groupId>fish.payara.maven.plugins</groupId>
                <artifactId>payara-micro-maven-plugin</artifactId>
                <version>1.0.0</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>bundle</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
```

**Running `mvn package`:**
```
[INFO] --- payara-micro-maven-plugin:1.0.0:bundle (default) @ some-project ---
[INFO] Configured Artifact: fish.payara.extras:payara-micro:4.1.1.171:jar
[INFO] Unpacking C:\Users\user\.m2\repository\fish\payara\extras\payara-micro\4.1.1.171\payara-micro-4.1.1.171.jar to C:\some-project\target\extracted-payaramicro with includes "" and excludes ""
[INFO] Configured Artifact: com.some-project:some-project:0.1-SNAPSHOT:war
[INFO] Copying 0 resource
[INFO] Copying 0 resource
[INFO] Copying some-project.war to C:\some-project\target\extracted-payaramicro\MICRO-INF\deploy\some-project-0.1-SNAPSHOT.war
[INFO] Building jar: C:\some-project\target\some-project-microbundle.jar
```

**Expected result:**
```
[INFO] Copying some-project.war to C:\some-project\target\extracted-payaramicro\MICRO-INF\deploy\some-project.war
```

When this microbundle is started the WAR will be available at `host:port/some-project-0.1-SNAPSHOT` instead of `host:port/some-project`.

## Further analysis

Seems the problem is the way the payara-micro-maven-plugin's bundle goal is [implemented](https://github.com/payara/maven-plugins/blob/master/payara-micro-maven-plugin/src/main/java/fish/payara/maven/plugins/micro/processor/ArtifactDeployProcessor.java#L57): 

```
...
executeMojo(dependencyPlugin,
		goal("copy"),
		configuration(
				element(name("artifactItems"),
						element(name("artifactItem"),
								element("groupId", "${project.groupId}"),
								element("artifactId", "${project.artifactId}"),
								element("version", "${project.version}"),
								element("type", "${project.packaging}")
						)
				),
				element(name("outputDirectory"), OUTPUT_FOLDER + MICROINF_DEPLOY_FOLDER)
		),
		environment
);
...
```

A possible solution is to use the `maven-resources-plugin:copy-resources` goal instead. Going to prepare a merge request implementing this.

**Comment by Skare69** [comment]
Pull request with fix here: https://github.com/payara/maven-plugins/pull/14.

**Comment by smillidge** [comment]
Fixed by #12
